src/rebar_state.erl

```diff
@@ -496,9 +496,6 @@ create_logic_providers(ProviderModules, State0) ->
     try
         lists:foldl(fun(ProviderMod, StateAcc) ->
                             case providers:new(ProviderMod, StateAcc) of
-                                {error, {Mod, Error}} ->
-                                    ?WARN("~ts", [Mod:format_error(Error)]),
-                                    StateAcc;
```

Comment:
I'd rather keep this one here. It's defensive for plugins that might go outside of the spec and what dialyzer permits and we'll instead hard crash rather than reporting well.

Reply:
What about if I add this to `providers:new`? `dialyzer` only complains because it's not there.

Reply:
you can, or we silence it I guess.

Reply:
Silenced.